server/record_test.go

```diff
@@ -0,0 +1,100 @@
+// +build record
+
+/*
+ * This file is part of Chihaya.
+ *
+ * Chihaya is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chihaya is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Chihaya.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package server
+
+import (
+	"bufio"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+type Record struct {
+	tid, uid  uint64
+	up, down  int64
+	absup     uint64
+	event, ip string
+}
+
+func randomString(len int) string {
+	bytes := make([]byte, len)
+	for i := 0; i < len; i++ {
+		bytes[i] = byte(65 + rand.Intn(90-65))
+	}
+	return string(bytes)
+}
+
+func randomIp() string {
+	return strconv.Itoa(rand.Intn(256)) + "." + strconv.Itoa(rand.Intn(256)) +
```

Comment:
This might create invalid IPs, such as 255.0.0.255, 0.0.0.0 or 0.45.1.125. I don't think it's cause for concern however as record treats IP as string.

Reply:
Same as with events, unsure if there's a merit to doing it considering the value passed is string anyway.